web/src/compose.js

```diff
@@ -872,3 +841,95 @@ function schedule_message_to_custom_date() {
+
+    const available_providers = page_params.realm_available_video_chat_providers;
+
+    if (
+        available_providers.zoom &&
+        page_params.realm_video_chat_provider === available_providers.zoom.id
+    ) {
+        if (is_audio_call) {
+            // TODO: Add support for generating audio-only Zoom calls here.
+            return;
+        }
+        abort_video_callbacks(edit_message_id);
+        const key = edit_message_id || "";
+
+        const make_zoom_call = () => {
+            video_call_xhrs.set(
+                key,
+                channel.post({
+                    url: "/json/calls/zoom/create",
+                    success(res) {
+                        video_call_xhrs.delete(key);
+                        insert_video_call_url(res.url, $target_textarea);
+                    },
+                    error(xhr, status) {
+                        video_call_xhrs.delete(key);
+                        if (
+                            status === "error" &&
+                            xhr.responseJSON &&
+                            xhr.responseJSON.code === "INVALID_ZOOM_TOKEN"
+                        ) {
+                            page_params.has_zoom_token = false;
+                        }
+                        if (status !== "abort") {
+                            ui_report.generic_embed_error(
+                                $t_html({defaultMessage: "Failed to create video call."}),
+                            );
+                        }
+                    },
+                }),
+            );
+        };
+
+        if (page_params.has_zoom_token) {
+            make_zoom_call();
+        } else {
+            zoom_token_callbacks.set(key, make_zoom_call);
+            window.open(
+                window.location.protocol + "//" + window.location.host + "/calls/zoom/register",
+                "_blank",
+                "width=800,height=500,noopener,noreferrer",
+            );
+        }
+    } else if (
+        available_providers.big_blue_button &&
+        page_params.realm_video_chat_provider === available_providers.big_blue_button.id
+    ) {
+        if (is_audio_call) {
+            // TODO: Add support for audio-only BigBlueButton calls here.
+            return;
+        }
+        const meeting_name = get_recipient_label() + " meeting";
+        channel.get({
+            url: "/json/calls/bigbluebutton/create",
+            data: {
+                meeting_name,
+            },
+            success(response) {
+                insert_video_call_url(response.url, $target_textarea);
+            },
+        });
+    } else {
+        // TODO: Use `new URL` to generate the URLs here.
+        const video_call_id = util.random_int(100000000000000, 999999999999999);
+        const video_call_link = page_params.jitsi_server_url + "/" + video_call_id;
+        if (is_audio_call) {
+            insert_audio_call_url(
+                video_call_link + "#config.startWithAudioMuted=true",
```

Comment:
This disables audio, not video. I think this bug is a good reminder that you should always test code end-to-end -- if you don't do that, you're going to have bugs like this where a feature doesn't do the main thing it's supposed to...

Reply:
Yeah, I agree that I messed things up here.